### ISSUE TITLE
Squiz ControlSignatureSniff hard-codes opener type while fixing

### ISSUE THREAD

**Issue by desaiuditd** [squizlabs/PHP_CodeSniffer#603]
```bash
test-theme|develop ⇒ ./library/vendors/composer/squizlabs/php_codesniffer/scripts/phpcbf --standard="WordPress-Core" page-templates/partials/testimonial.php --vvv      
Changing into directory /Users/udit/test-theme/page-templates/partials
Processing testimonial.php [PHP => 304 tokens in 43 lines]... DONE in 6ms (47 fixable violations)
        => Fixing file: 12/47 violations remaining [made 2 passes]... PHP Notice:  Undefined index: scope_condition in /Users/udit/test-theme/library/vendors/composer/squizlabs/php_codesniffer/CodeSniffer/Standards/Generic/Sniffs/WhiteSpace/ScopeIndentSniff.php on line 417
PHP Notice:  Undefined index:  in /Users/udit/test-theme/library/vendors/composer/squizlabs/php_codesniffer/CodeSniffer/File.php on line 3464
PHP Notice:  Undefined index:  in /Users/udit/test-theme/library/vendors/composer/squizlabs/php_codesniffer/CodeSniffer/File.php on line 3464
PHP Notice:  Undefined index:  in /Users/udit/test-theme/library/vendors/composer/squizlabs/php_codesniffer/CodeSniffer/File.php on line 3476
PHP Notice:  Undefined index:  in /Users/udit/test-theme/library/vendors/composer/squizlabs/php_codesniffer/CodeSniffer/File.php on line 3464
```

These errors occurs in thousands of number and doesn't stop.

PHPCS Version
```bash
test-theme|develop ⇒ ./library/vendors/composer/squizlabs/php_codesniffer/scripts/phpcbf --version                                                                
PHP_CodeSniffer version 2.3.2 (stable) by Squiz (http://www.squiz.net)
```

Any idea ?

**Comment by JDGrimes** [comment]
Example code:

```
<?php if ( $foo ): ?>
<?php endif; ?>
```

**Comment by desaiuditd** [comment]
Same problem with the example code as well.

**Comment by stephenharris** [comment]
Just to add a "me too" (2.3.2, WPCS 0.5.0). It seems the `Generic.WhiteSpace.ScopeIndent` is responsible (as mentioned in the WPCS issue). The file in question contained `while`,`if` and `foreach` control structures using the `:`-syntax. Changing these and `phpcbf` runs without any problems.

**Comment by gsherwood** [comment]
I can't replicate this error with the code snippet provided, also using version 2.3.2. Here is the full `-vvv` output of my test. Can you please run the same command and paste what you get when processing that sample code snippet.

```
$ phpcs temp.php --report=diff -vvv
Processing ruleset /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/ruleset.xml
	Adding sniff files from "/.../PEAR/Sniffs/" directory
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Classes/ClassDeclarationSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Commenting/ClassCommentSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Commenting/FileCommentSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Commenting/FunctionCommentSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Commenting/InlineCommentSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/ControlStructures/ControlSignatureSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/ControlStructures/MultiLineConditionSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Files/IncludingFileSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Formatting/MultiLineAssignmentSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Functions/FunctionCallSignatureSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Functions/FunctionDeclarationSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/Functions/ValidDefaultValueSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/NamingConventions/ValidClassNameSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/NamingConventions/ValidFunctionNameSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/NamingConventions/ValidVariableNameSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/WhiteSpace/ObjectOperatorIndentSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/WhiteSpace/ScopeClosingBraceSniff.php
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/PEAR/Sniffs/WhiteSpace/ScopeIndentSniff.php
	Processing rule "Generic.Functions.FunctionCallArgumentSpacing"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/Functions/FunctionCallArgumentSpacingSniff.php
	Processing rule "Generic.NamingConventions.UpperCaseConstantName"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/NamingConventions/UpperCaseConstantNameSniff.php
	Processing rule "Generic.PHP.LowerCaseConstant"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/PHP/LowerCaseConstantSniff.php
	Processing rule "Generic.PHP.DisallowShortOpenTag"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/PHP/DisallowShortOpenTagSniff.php
	Processing rule "Generic.WhiteSpace.DisallowTabIndent"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/WhiteSpace/DisallowTabIndentSniff.php
	Processing rule "Generic.Commenting.DocComment"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/Commenting/DocCommentSniff.php
	Processing rule "Generic.Files.LineLength"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/Files/LineLengthSniff.php
		=> property "lineLimit" set to "85"
		=> property "absoluteLineLimit" set to "0"
	Processing rule "Generic.Files.LineEndings"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/Files/LineEndingsSniff.php
		=> property "eolChar" set to "\n"
	Processing rule "Generic.Functions.FunctionCallArgumentSpacing.TooMuchSpaceAfterComma"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/Functions/FunctionCallArgumentSpacingSniff.php
		=> severity set to 0
	Processing rule "Generic.ControlStructures.InlineControlStructure"
		=> /usr/local/PEAR/PHP/CodeSniffer/Standards/Generic/Sniffs/ControlStructures/InlineControlStructureSniff.php
		=> property "error" set to "false"
=> Ruleset processing complete; included 27 sniffs and excluded 0
Registered PEAR_Sniffs_Classes_ClassDeclarationSniff
Registered PEAR_Sniffs_Commenting_ClassCommentSniff
Registered PEAR_Sniffs_Commenting_FileCommentSniff
Registered PEAR_Sniffs_Commenting_FunctionCommentSniff
Registered PEAR_Sniffs_Commenting_InlineCommentSniff
Registered PEAR_Sniffs_ControlStructures_ControlSignatureSniff
Registered PEAR_Sniffs_ControlStructures_MultiLineConditionSniff
Registered PEAR_Sniffs_Files_IncludingFileSniff
Registered PEAR_Sniffs_Formatting_MultiLineAssignmentSniff
Registered PEAR_Sniffs_Functions_FunctionCallSignatureSniff
Registered PEAR_Sniffs_Functions_FunctionDeclarationSniff
Registered PEAR_Sniffs_Functions_ValidDefaultValueSniff
Registered PEAR_Sniffs_NamingConventions_ValidClassNameSniff
Registered PEAR_Sniffs_NamingConventions_ValidFunctionNameSniff
Registered PEAR_Sniffs_NamingConventions_ValidVariableNameSniff
Registered PEAR_Sniffs_WhiteSpace_ObjectOperatorIndentSniff
Registered PEAR_Sniffs_WhiteSpace_ScopeClosingBraceSniff
Registered PEAR_Sniffs_WhiteSpace_ScopeIndentSniff
Registered Generic_Sniffs_Functions_FunctionCallArgumentSpacingSniff
Registered Generic_Sniffs_NamingConventions_UpperCaseConstantNameSniff
Registered Generic_Sniffs_PHP_LowerCaseConstantSniff
Registered Generic_Sniffs_PHP_DisallowShortOpenTagSniff
Registered Generic_Sniffs_WhiteSpace_DisallowTabIndentSniff
Registered Generic_Sniffs_Commenting_DocCommentSniff
Registered Generic_Sniffs_Files_LineLengthSniff
Registered Generic_Sniffs_Files_LineEndingsSniff
Registered Generic_Sniffs_ControlStructures_InlineControlStructureSniff
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_DO => do
	Process token [2]: T_WHITESPACE => ·
	Process token  3 : T_OPEN_CURLY_BRACKET => {
	Process token [4]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_WHITESPACE => ·
	Process token [2]: T_WHILE => while
	Process token [3]: T_WHITESPACE => ·
	Process token  4 : T_OPEN_PARENTHESIS => (
	Process token [5]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token  1 : T_SEMICOLON => ;
	Process token [2]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_WHILE => while
	Process token [2]: T_WHITESPACE => ·
	Process token  3 : T_OPEN_PARENTHESIS => (
	Process token [4]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_WHITESPACE => ·
	Process token  2 : T_OPEN_CURLY_BRACKET => {
	Process token [3]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_FOR => for
	Process token [2]: T_WHITESPACE => ·
	Process token  3 : T_OPEN_PARENTHESIS => (
	Process token [4]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_WHITESPACE => ·
	Process token  2 : T_OPEN_CURLY_BRACKET => {
	Process token [3]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_IF => if
	Process token [2]: T_WHITESPACE => ·
	Process token  3 : T_OPEN_PARENTHESIS => (
	Process token [4]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_WHITESPACE => ·
	Process token  2 : T_OPEN_CURLY_BRACKET => {
	Process token [3]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_FOREACH => foreach
	Process token [2]: T_WHITESPACE => ·
	Process token  3 : T_OPEN_PARENTHESIS => (
	Process token [4]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_WHITESPACE => ·
	Process token  2 : T_OPEN_CURLY_BRACKET => {
	Process token [3]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token  1 : T_CLOSE_CURLY_BRACKET => }
	Process token [2]: T_WHITESPACE => ·
	Process token [3]: T_ELSE => else
	Process token [4]: T_WHITESPACE => ·
	Process token [5]: T_IF => if
	Process token [6]: T_WHITESPACE => ·
	Process token  7 : T_OPEN_PARENTHESIS => (
	Process token [8]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_WHITESPACE => ·
	Process token  2 : T_OPEN_CURLY_BRACKET => {
	Process token [3]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token  1 : T_CLOSE_CURLY_BRACKET => }
	Process token [2]: T_WHITESPACE => ·
	Process token [3]: T_ELSEIF => elseif
	Process token [4]: T_WHITESPACE => ·
	Process token  5 : T_OPEN_PARENTHESIS => (
	Process token [6]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_WHITESPACE => ·
	Process token  2 : T_OPEN_CURLY_BRACKET => {
	Process token [3]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token  1 : T_CLOSE_CURLY_BRACKET => }
	Process token [2]: T_WHITESPACE => ·
	Process token [3]: T_ELSE => else
	Process token [4]: T_WHITESPACE => ·
	Process token  5 : T_OPEN_CURLY_BRACKET => {
	Process token [6]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_DO => do
	Process token [2]: T_WHITESPACE => ·
	Process token  3 : T_OPEN_CURLY_BRACKET => {
	Process token [4]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_CLOSE_TAG => ?>
	*** END PHP TOKENIZING ***
Creating file list... DONE (1 files in queue)
Changing into directory /Users/gsherwood/Sites/Projects/PHP_CodeSniffer
Processing temp.php
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_IF => if
	Process token [2]: T_WHITESPACE => ·
	Process token  3 : T_OPEN_PARENTHESIS => (
	Process token [4]: T_WHITESPACE => ·
	Process token [5]: T_VARIABLE => $foo
	Process token [6]: T_WHITESPACE => ·
	Process token  7 : T_CLOSE_PARENTHESIS => )
	Process token  8 : T_COLON => :
	Process token [9]: T_WHITESPACE => ·
	Process token [10]: T_CLOSE_TAG => ?>\n
	Process token [11]: T_OPEN_TAG => <?php·
	Process token [12]: T_ENDIF => endif
	Process token  13 : T_SEMICOLON => ;
	Process token [14]: T_WHITESPACE => ·
	Process token [15]: T_CLOSE_TAG => ?>\n
	*** END PHP TOKENIZING ***
	*** START TOKEN MAP ***
	*** END TOKEN MAP ***
	*** START SCOPE MAP ***
	Start scope map at 1:T_IF => if
	*** START SCOPE MAP ***
	Process token 2 on line 1 []: T_WHITESPACE => ·
	Process token 3 on line 1 []: T_OPEN_PARENTHESIS => (
	Process token 4 on line 1 []: T_WHITESPACE => ·
	Process token 5 on line 1 []: T_VARIABLE => $foo
	Process token 6 on line 1 []: T_WHITESPACE => ·
	Process token 7 on line 1 []: T_CLOSE_PARENTHESIS => )
	Process token 8 on line 1 []: T_COLON => :
	=> Found scope opener for 1:T_IF
	Process token 9 on line 1 [opener:8;]: T_WHITESPACE => ·
	Process token 10 on line 1 [opener:8;]: T_CLOSE_TAG => ?>\n
	Process token 11 on line 2 [opener:8;]: T_OPEN_TAG => <?php·
	Process token 12 on line 2 [opener:8;]: T_ENDIF => endif
	=> Found scope closer (12:T_ENDIF) for 1:T_IF
	*** END SCOPE MAP ***
	*** START LEVEL MAP ***
	Process token 0 on line 1 [col:1;len:6;lvl:0;]: T_OPEN_TAG => <?php·
	Process token 1 on line 1 [col:7;len:2;lvl:0;]: T_IF => if
	Process token 2 on line 1 [col:9;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 3 on line 1 [col:10;len:1;lvl:0;]: T_OPEN_PARENTHESIS => (
	Process token 4 on line 1 [col:11;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 5 on line 1 [col:12;len:4;lvl:0;]: T_VARIABLE => $foo
	Process token 6 on line 1 [col:16;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 7 on line 1 [col:17;len:1;lvl:0;]: T_CLOSE_PARENTHESIS => )
	Process token 8 on line 1 [col:18;len:1;lvl:0;]: T_COLON => :
	=> Found scope opener for 1:T_IF
		* level increased *
		* token 1:T_IF added to conditions array *
		Process token 9 on line 1 [col:19;len:1;lvl:1;conds;T_IF;]: T_WHITESPACE => ·
		Process token 10 on line 1 [col:20;len:2;lvl:1;conds;T_IF;]: T_CLOSE_TAG => ?>\n
		Process token 11 on line 2 [col:1;len:6;lvl:1;conds;T_IF;]: T_OPEN_TAG => <?php·
		Process token 12 on line 2 [col:7;len:5;lvl:1;conds;T_IF;]: T_ENDIF => endif
		=> Found scope closer for 8:T_COLON
		* token T_IF removed from conditions array *
		* level decreased *
	Process token 13 on line 2 [col:12;len:1;lvl:0;]: T_SEMICOLON => ;
	Process token 14 on line 2 [col:13;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 15 on line 2 [col:14;len:2;lvl:0;]: T_CLOSE_TAG => ?>\n
	*** END LEVEL MAP ***
	*** START ADDITIONAL PHP PROCESSING ***
	*** END ADDITIONAL PHP PROCESSING ***
[PHP => 16 tokens in 2 lines]...
	*** START TOKEN PROCESSING ***
		Process token 0: T_OPEN_TAG => <?php·
			Processing PEAR_Sniffs_Commenting_FileCommentSniff... DONE in 0 seconds
			Processing PEAR_Sniffs_WhiteSpace_ScopeIndentSniff... DONE in 0.0001 seconds
			Processing Generic_Sniffs_PHP_DisallowShortOpenTagSniff... DONE in 0 seconds
			Processing Generic_Sniffs_WhiteSpace_DisallowTabIndentSniff... DONE in 0 seconds
			Processing Generic_Sniffs_Files_LineLengthSniff... DONE in 0 seconds
			Processing Generic_Sniffs_Files_LineEndingsSniff... DONE in 0 seconds
		Process token 1: T_IF => if
			Processing PEAR_Sniffs_ControlStructures_ControlSignatureSniff... DONE in 0.0001 seconds
			Processing PEAR_Sniffs_ControlStructures_MultiLineConditionSniff... DONE in 0.0001 seconds
			Processing PEAR_Sniffs_WhiteSpace_ScopeClosingBraceSniff... DONE in 0 seconds
			Processing Generic_Sniffs_ControlStructures_InlineControlStructureSniff... DONE in 0 seconds
		Process token 2: T_WHITESPACE => ·
		Process token 3: T_OPEN_PARENTHESIS => (
		Process token 4: T_WHITESPACE => ·
		Process token 5: T_VARIABLE => $foo
			Processing PEAR_Sniffs_NamingConventions_ValidVariableNameSniff... DONE in 0 seconds
		Process token 6: T_WHITESPACE => ·
		Process token 7: T_CLOSE_PARENTHESIS => )
		Process token 8: T_COLON => :
		Process token 9: T_WHITESPACE => ·
		Process token 10: T_CLOSE_TAG => ?>\n
		Process token 11: T_OPEN_TAG => <?php·
			Processing Generic_Sniffs_PHP_DisallowShortOpenTagSniff... DONE in 0 seconds
		Process token 12: T_ENDIF => endif
		Process token 13: T_SEMICOLON => ;
		Process token 14: T_WHITESPACE => ·
		Process token 15: T_CLOSE_TAG => ?>\n
	*** END TOKEN PROCESSING ***
	*** START SNIFF PROCESSING REPORT ***
	PEAR_Sniffs_WhiteSpace_ScopeIndentSniff: 0.0001 secs
	PEAR_Sniffs_ControlStructures_ControlSignatureSniff: 0.0001 secs
	PEAR_Sniffs_ControlStructures_MultiLineConditionSniff: 0.0001 secs
	PEAR_Sniffs_Commenting_FileCommentSniff: 0 secs
	PEAR_Sniffs_WhiteSpace_ScopeClosingBraceSniff: 0 secs
	Generic_Sniffs_Files_LineLengthSniff: 0 secs
	Generic_Sniffs_WhiteSpace_DisallowTabIndentSniff: 0 secs
	Generic_Sniffs_PHP_DisallowShortOpenTagSniff: 0 secs
	PEAR_Sniffs_NamingConventions_ValidVariableNameSniff: 0 secs
	Generic_Sniffs_Files_LineEndingsSniff: 0 secs
	Generic_Sniffs_ControlStructures_InlineControlStructureSniff: 0 secs
	*** END SNIFF PROCESSING REPORT ***
DONE in 2ms (4 errors, 0 warnings)
	*** START FILE FIXING ***
---START FILE CONTENT---
1|<?php if ( $foo ): ?>
2|<?php endif; ?>
3|
--- END FILE CONTENT ---
	E: [Line 1] First condition of a multi-line IF statement must directly follow the opening parenthesis (PEAR.ControlStructures.MultiLineCondition.SpacingAfterOpenBrace)
	PEAR_Sniffs_ControlStructures_MultiLineConditionSniff (line 95) replaced token 4 (T_WHITESPACE) "·$foo" => "$foo"
	E: [Line 1] There must be a single space between the closing parenthesis and the opening brace of a multi-line IF statement; found 0 spaces (PEAR.ControlStructures.MultiLineCondition.SpaceBeforeOpenBrace)
	PEAR_Sniffs_ControlStructures_MultiLineConditionSniff (line 261) replaced token 7 (T_CLOSE_PARENTHESIS) ")" => ")·"
	E: [Line 2] Closing brace must be on a line by itself (PEAR.WhiteSpace.ScopeClosingBrace.Line)
	PEAR_Sniffs_WhiteSpace_ScopeClosingBraceSniff (line 116) replaced token 12 (T_ENDIF) "endif" => "\nendif"
	* fixed 3 violations, starting loop 2 *
---START FILE CONTENT---
1|<?php if ($foo ) : ?>
2|<?php
3|endif; ?>
4|
--- END FILE CONTENT ---
	E: [Line 3] Closing brace indented incorrectly; expected 6 spaces, found 0 (PEAR.WhiteSpace.ScopeClosingBrace.Indent)
	PEAR_Sniffs_WhiteSpace_ScopeClosingBraceSniff (line 154) replaced token 13 (T_ENDIF) "endif" => "······endif"
	* fixed 1 violations, starting loop 3 *
---START FILE CONTENT---
1|<?php if ($foo ) : ?>
2|<?php
3|      endif; ?>
4|
--- END FILE CONTENT ---
	*** END FILE FIXING ***
--- temp.php
+++ PHP_CodeSniffer
@@ -1,2 +1,3 @@
-<?php if ( $foo ): ?>
-<?php endif; ?>
+<?php if ($foo ) : ?>
+<?php
+      endif; ?>
```

**Comment by stephenharris** [comment]
I believe you're using the PEAR ruleset which doesn't have the (seemingly) offending rule `Generic.WhiteSpace.ScopeIndent`. Incidently, this *alone* is not enough to replicate the issue, i.e:

```
<?xml version="1.0"?>
<ruleset name="Test">
	<description>A custom ruleset to debug #603</description>

	<!--
	We may also want to to include all the rules in a standard
	-->
	<rule ref="Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis"/>
	
	<rule ref="Generic.WhiteSpace.ScopeIndent">
		<properties>
			<property name="indent" value="4"/>
			<property name="tabIndent" value="true"/>
		</properties>
	</rule>
</ruleset>
```

fixes the example snippet without any issue. Clearly there is more factors at play here. I'll post back if I'm able to determine what any of these are.

**Comment by gsherwood** [comment]
> I believe you're using the PEAR ruleset which doesn't have the (seemingly) offending rule Generic.WhiteSpace.ScopeIndent.

I tried with both PEAR and with the WordPress standards (and every other one actually) and couldn't replicate it. Also, PEAR does include ScopeIndent as well. It just wraps it to override a single rule (https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/PEAR/Sniffs/WhiteSpace/ScopeIndentSniff.php), but the main code still runs the same. So the code is PEAR.WhiteSpace.ScopeIndent.

**Comment by JDGrimes** [comment]
I don't get the issue when using the `PEAR` standard, but I do when running `WordPress-Core`. So it's possible that it is an issue with one of our custom sniffs after all.

**Comment by JDGrimes** [comment]
@stephenharris You could try copying the `WordPress-Core` sniff and removing the rules one-by-one until the problem goes away.

**Comment by stephenharris** [comment]
Here's the smallest ruleset I could get to replicate this issue:

	<rule ref="Squiz.ControlStructures.ControlSignature.NewlineAfterOpenBrace">
		<severity>0</severity>
	</rule>

	<rule ref="Generic.WhiteSpace.ScopeIndent">
		<properties>
			<property name="indent" value="4"/>
			<property name="tabIndent" value="true"/>
		</properties>
	</rule>

	<rule ref="WordPress.WhiteSpace.ControlStructureSpacing"/>

I'm inclined to agree with @JDGrimes but for what it's worth I've posted below the result when I ran `--report=diff -vvv` (up to the first PHP error).

```
	Processing rule "Squiz.ControlStructures.ControlSignature.NewlineAfterOpenBrace"
		=> /var/www/html/sandbox/vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/Squiz/Sniffs/ControlStructures/ControlSignatureSniff.php
		=> severity set to 0
	Processing rule "Generic.WhiteSpace.ScopeIndent"
		=> /var/www/html/sandbox/vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/Generic/Sniffs/WhiteSpace/ScopeIndentSniff.php
		=> property "indent" set to "4"
		=> property "tabIndent" set to "true"
	Processing rule "WordPress.WhiteSpace.ControlStructureSpacing"
		=> /var/www/html/sandbox/vendor/wp-coding-standards/wpcs/WordPress/Sniffs/WhiteSpace/ControlStructureSpacingSniff.php
=> Ruleset processing complete; included 3 sniffs and excluded 0
Registered Squiz_Sniffs_ControlStructures_ControlSignatureSniff
Registered Generic_Sniffs_WhiteSpace_ScopeIndentSniff
Registered WordPress_Sniffs_WhiteSpace_ControlStructureSpacingSniff
Creating file list... DONE (1 files in queue)
Changing into directory /var/www/html/sandbox
Processing test.php 
	*** START PHP TOKENIZING ***
	Process token [0]: T_OPEN_TAG => <?php·
	Process token [1]: T_IF => if
	Process token [2]: T_WHITESPACE => ·
	Process token  3 : T_OPEN_PARENTHESIS => (
	Process token [4]: T_WHITESPACE => ·
	Process token [5]: T_VARIABLE => $foo
	Process token [6]: T_WHITESPACE => ·
	Process token  7 : T_CLOSE_PARENTHESIS => )
	Process token  8 : T_COLON => :
	Process token [9]: T_WHITESPACE => ···
	Process token [10]: T_CLOSE_TAG => ?>\n
	Process token [11]: T_OPEN_TAG => <?php·
	Process token [12]: T_ENDIF => endif
	Process token  13 : T_SEMICOLON => ;
	Process token [14]: T_WHITESPACE => ·
	Process token [15]: T_CLOSE_TAG => ?>\n
	*** END PHP TOKENIZING ***
	*** START TOKEN MAP ***
	*** END TOKEN MAP ***
	*** START SCOPE MAP ***
	Start scope map at 1:T_IF => if
	*** START SCOPE MAP ***
	Process token 2 on line 1 []: T_WHITESPACE => ·
	Process token 3 on line 1 []: T_OPEN_PARENTHESIS => (
	Process token 4 on line 1 []: T_WHITESPACE => ·
	Process token 5 on line 1 []: T_VARIABLE => $foo
	Process token 6 on line 1 []: T_WHITESPACE => ·
	Process token 7 on line 1 []: T_CLOSE_PARENTHESIS => )
	Process token 8 on line 1 []: T_COLON => :
	=> Found scope opener for 1:T_IF
	Process token 9 on line 1 [opener:8;]: T_WHITESPACE => ···
	Process token 10 on line 1 [opener:8;]: T_CLOSE_TAG => ?>\n
	Process token 11 on line 2 [opener:8;]: T_OPEN_TAG => <?php·
	Process token 12 on line 2 [opener:8;]: T_ENDIF => endif
	=> Found scope closer (12:T_ENDIF) for 1:T_IF
	*** END SCOPE MAP ***
	*** START LEVEL MAP ***
	Process token 0 on line 1 [col:1;len:6;lvl:0;]: T_OPEN_TAG => <?php·
	Process token 1 on line 1 [col:7;len:2;lvl:0;]: T_IF => if
	Process token 2 on line 1 [col:9;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 3 on line 1 [col:10;len:1;lvl:0;]: T_OPEN_PARENTHESIS => (
	Process token 4 on line 1 [col:11;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 5 on line 1 [col:12;len:4;lvl:0;]: T_VARIABLE => $foo
	Process token 6 on line 1 [col:16;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 7 on line 1 [col:17;len:1;lvl:0;]: T_CLOSE_PARENTHESIS => )
	Process token 8 on line 1 [col:18;len:1;lvl:0;]: T_COLON => :
	=> Found scope opener for 1:T_IF
		* level increased *
		* token 1:T_IF added to conditions array *
		Process token 9 on line 1 [col:19;len:3;lvl:1;conds;T_IF;]: T_WHITESPACE => ···
		Process token 10 on line 1 [col:22;len:2;lvl:1;conds;T_IF;]: T_CLOSE_TAG => ?>\n
		Process token 11 on line 2 [col:1;len:6;lvl:1;conds;T_IF;]: T_OPEN_TAG => <?php·
		Process token 12 on line 2 [col:7;len:5;lvl:1;conds;T_IF;]: T_ENDIF => endif
		=> Found scope closer for 8:T_COLON
		* token T_IF removed from conditions array *
		* level decreased *
	Process token 13 on line 2 [col:12;len:1;lvl:0;]: T_SEMICOLON => ;
	Process token 14 on line 2 [col:13;len:1;lvl:0;]: T_WHITESPACE => ·
	Process token 15 on line 2 [col:14;len:2;lvl:0;]: T_CLOSE_TAG => ?>\n
	*** END LEVEL MAP ***
	*** START ADDITIONAL PHP PROCESSING ***
	*** END ADDITIONAL PHP PROCESSING ***
[PHP => 16 tokens in 2 lines]... 
	*** START TOKEN PROCESSING ***
		Process token 0: T_OPEN_TAG => <?php·
			Processing Generic_Sniffs_WhiteSpace_ScopeIndentSniff... DONE in 0.0004 seconds
		Process token 1: T_IF => if
			Processing Squiz_Sniffs_ControlStructures_ControlSignatureSniff... DONE in 0.0002 seconds
			Processing WordPress_Sniffs_WhiteSpace_ControlStructureSpacingSniff... DONE in 0.0001 seconds
		Process token 2: T_WHITESPACE => ·
		Process token 3: T_OPEN_PARENTHESIS => (
		Process token 4: T_WHITESPACE => ·
		Process token 5: T_VARIABLE => $foo
		Process token 6: T_WHITESPACE => ·
		Process token 7: T_CLOSE_PARENTHESIS => )
		Process token 8: T_COLON => :
		Process token 9: T_WHITESPACE => ···
		Process token 10: T_CLOSE_TAG => ?>\n
		Process token 11: T_OPEN_TAG => <?php·
		Process token 12: T_ENDIF => endif
		Process token 13: T_SEMICOLON => ;
		Process token 14: T_WHITESPACE => ·
		Process token 15: T_CLOSE_TAG => ?>\n
	*** END TOKEN PROCESSING ***
	*** START SNIFF PROCESSING REPORT ***
	Generic_Sniffs_WhiteSpace_ScopeIndentSniff: 0.0004 secs
	Squiz_Sniffs_ControlStructures_ControlSignatureSniff: 0.0002 secs
	WordPress_Sniffs_WhiteSpace_ControlStructureSpacingSniff: 0.0001 secs
	*** END SNIFF PROCESSING REPORT ***
DONE in 12ms (2 errors, 0 warnings)
	*** START FILE FIXING ***
---START FILE CONTENT---
1|<?php if ( $foo ):   ?>
2|<?php endif; ?>
3|
--- END FILE CONTENT ---
	E: [Line 1] Expected 1 space after closing parenthesis; found "" (Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis)
	Squiz_Sniffs_ControlStructures_ControlSignatureSniff (line 118) replaced token 7 (T_CLOSE_PARENTHESIS) ")" => ")·"
	E: [Line 1] Space between opening control structure and T_COLON is required (WordPress.WhiteSpace.ControlStructureSpacing.NoSpaceBetweenStructureColon)
	=> Changeset started by WordPress_Sniffs_WhiteSpace_ControlStructureSpacingSniff (line 128)
		Q: WordPress_Sniffs_WhiteSpace_ControlStructureSpacingSniff (line 129) replaced token 8 (T_COLON) ":" => "·:"
		A: WordPress_Sniffs_WhiteSpace_ControlStructureSpacingSniff (line 130) replaced token 8 (T_COLON) ":" => "·:"
	=> Changeset ended: 1 changes applied
	* fixed 2 violations, starting loop 2 *
---START FILE CONTENT---
1|<?php if ( $foo )  :   ?>
2|<?php endif; ?>
3|
--- END FILE CONTENT ---
	E: [Line 1] Expected 1 space after closing parenthesis; found "  " (Squiz.ControlStructures.ControlSignature.SpaceAfterCloseParenthesis)
	=> Changeset started by Squiz_Sniffs_ControlStructures_ControlSignatureSniff (line 120)
		Q: Squiz_Sniffs_ControlStructures_ControlSignatureSniff (line 121) replaced token 7 (T_CLOSE_PARENTHESIS) ")" => ")·{"
		Q: Squiz_Sniffs_ControlStructures_ControlSignatureSniff (line 122) replaced token 9 (T_COLON) ":" => ""
		A: Squiz_Sniffs_ControlStructures_ControlSignatureSniff (line 133) replaced token 7 (T_CLOSE_PARENTHESIS) ")" => ")·{"
		A: Squiz_Sniffs_ControlStructures_ControlSignatureSniff (line 133) replaced token 9 (T_COLON) ":" => ""
	=> Changeset ended: 2 changes applied
	* fixed 2 violations, starting loop 3 *
---START FILE CONTENT---
1|<?php if ( $foo ) {     ?>
2|<?php endif; ?>
3|
--- END FILE CONTENT ---
PHP Notice:  Undefined index: scope_condition in /var/www/html/sandbox/vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/Generic/Sniffs/WhiteSpace/ScopeIndentSniff.php on line 417
```

**Comment by gsherwood** [comment]
I just tried replicating this again and couldn't. Then I realised that my copy of the WordPress coding standards repo was not updated (sorry, I thought I had done that). After after that repo, I can now replicate the errors.

I'll take a look and see if I can track down where they are coming from now.

**Comment by gsherwood** [comment]
Thanks a lot for that `-vvv` output; it made is easy to track down. The problem bit was this:
```
Q: Squiz_Sniffs_ControlStructures_ControlSignatureSniff (line 121) replaced token 7 (T_CLOSE_PARENTHESIS) ")" => ")·{"
Q: Squiz_Sniffs_ControlStructures_ControlSignatureSniff (line 122) replaced token 9 (T_COLON) ":" => ""
```

Where you can see that the Squiz sniff was replacing the colon opener with a curly brace opener. It had the opener type hard-coded if the fixing code because it assumed it was always part of the Squiz standard. Previously, this sniff used pattern matching and just ignored control structures with alternate syntax, but it was updated when auto-fixing came in and so is now more generically useful.

**Comment by stephenharris** [comment]
Just tested this, and can confirm this patch fixes the issue. Thank you!
